src/lib.rs

```diff
@@ -285,42 +318,78 @@ where
 /// # use eyre::Result;
 /// # async fn doc() -> eyre::Result<()> {
 /// let address = "127.0.0.1:3310";
-/// let mut clamd_client = ClamdClientBuilder::tcp_socket(address).build();
+/// let mut clamd_client = ClamdClientBuilder::tcp_socket(&address.parse()?).build();
 /// clamd_client.ping().await?;
 /// # Ok(())
 /// # }
 /// ```
-pub struct ClamdClient<T: ToSocketAddrs + ToOwned<Owned = T>> {
-    //codec: Framed<T, ClamdZeroDelimitedCodec>,
-    socket_type: SocketType<T>,
+pub struct ClamdClient {
+    socket_type: SocketType,
     connection_type: ConnectionType,
     chunk_size: usize,
+    socket: Option<Arc<Mutex<Framed<SocketWrapper, ClamdZeroDelimitedCodec>>>>,
```

Comment:
I think the Arc should be around the whole `ClamdClient` for the `Mutex` to make sense. With how the code is currently layed out, if you use the ClamdClient from multiple threads, it will set socket of the copy of each `ClamdClient` making the `Mutex` superfluous.

Reply:
Or we make the whole ClamdClient copyable and change the connect method to modify the mutex directly (also the Option should then be inside the Mutex)

Reply:
What do you think of the new implementation ?